channel.go

```diff
@@ -209,7 +210,13 @@ func (c *serialChannel) listen() (net.Listener, error) {
 func (c *serialChannel) teardown() error {
 	// wait for the session to be fully shutdown first
 	if c.waitCh != nil {
-		<-c.waitCh
+		t := time.NewTimer(channelCloseTimeout)
+		select {
+		case <-c.waitCh:
+			t.Stop()
```

Comment:
according with the doc

```go
// To prevent a timer created with NewTimer from firing after a call to Stop,
// check the return value and drain the channel.
// For example, assuming the program has not received from t.C already:
//
// 	if !t.Stop() {
// 		<-t.C
// 	}
//
```

Reply:
No, that's not necessary. That comment is to prevent accidental read from the timer channel but in our case the function returns and nothing is reading that channel.

Reply:
ok, make sense